ledger/api-server-damlonx/reference-v2/src/main/scala/com/daml/ledger/api/server/damlonx/reference/v2/ReferenceServer.scala

```diff
@@ -38,51 +38,71 @@ object ReferenceServer extends App {
 
   implicit val system: ActorSystem = ActorSystem("indexed-kvutils")
   implicit val materializer: Materializer = Materializer(system)
-  implicit val ec: ExecutionContext = system.dispatcher
-
-  val ledger = new InMemoryKVParticipantState(config.participantId)
-
-  val readService = ledger
-  val writeService = ledger
-  val authService = AuthServiceWildcard
+  implicit val executionContext: ExecutionContext = system.dispatcher
+
+  val resource = for {
+    _ <- ResourceOwner.forActorSystem(() => system).acquire()
```

Comment:
Would it make sense to pass arguments by name instead of by value for these helpers?

Reply:
I'll add a comment here. This is a hack to "take" ownership of something that's already been constructed. It shouldn't be necessary except that we want to use them implicitly all over the place.